core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -185,7 +186,27 @@ else if (spreadItemIndex == spreadEndIndex)
 
 	private int getSpreadIndex(ZonedDateTime atTime)
 	{
-		return (int) (Duration.between(getStartTime(), atTime).toMinutes() / 15);
+		int dstCorrection = determineDstCorrectionMinutes(atTime);
```

Comment:
Yes, that works, see this:
```
		ZonedDateTime dateTime = ZonedDateTime.of(2018, 10, 28, 0, 0, 0, 0, ZoneId.of("Europe/Amsterdam"));

		for (int i = 0; (i < 25); i++)
		{
			System.out.println(dateTime.plusHours(i).toLocalTime().getHour());
		}
```
This outputs:

```
0
1
2
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
```
Thanks! I'll update the code.

Reply:
We need to convert to ``LocalDateTime`` rather than ``LocalTime``, because otherwise we can't see the difference anymore between start of the current day and the next one. But with that it works and is easier to understand than my earlier solution.